src/downloadAzureProject/setupProjectFolder.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from '@azure/arm-appservice';
+import { RequestPrepareOptions } from '@azure/ms-rest-js';
+import * as extract from 'extract-zip';
+import * as querystring from 'querystring';
+import * as vscode from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { ProjectLanguage } from '../constants';
+import { GlobalStates } from '../extension';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { SlotTreeItemBase } from "../tree/SlotTreeItemBase";
+import { getNameFromId } from '../utils/azure';
+import { requestUtils } from '../utils/requestUtils';
+
+export async function setupProjectFolder(uri: vscode.Uri, vsCodeFilePathUri: vscode.Uri): Promise<void> {
+    const parsedQuery: querystring.ParsedUrlQuery = querystring.parse(uri.query);
+    const resourceId: string = getRequiredQueryParameter(parsedQuery, 'resourceId');
+    const devContainerName: string = getRequiredQueryParameter(parsedQuery, 'devcontainer');
+    const language: string = getRequiredQueryParameter(parsedQuery, 'language');
+    const downloadAppContent: string = getRequiredQueryParameter(parsedQuery, 'downloadAppContent');
+
+    ext.context.globalState.update(GlobalStates.projectLanguage, getProjectLanguageForLanguage(language));
+    const toBeDeletedFolderPathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, 'temp');
+
+    try {
+        const functionAppName: string = getNameFromId(resourceId);
+        const downloadFilePath: string = vscode.Uri.joinPath(toBeDeletedFolderPathUri, `${functionAppName}.zip`).fsPath;
+
+        vscode.window.showInformationMessage(localize('settingUpFunctionAppLocalProjInfoMessage', `Setting up project for function app '${functionAppName}' with language '${language}'.`));
+
+        if (downloadAppContent === 'true') {
+            // NOTE: We don't want to download app content for compiled languages.
```

Comment:
When customers create a compiled language functionApp from Azure Portal, we don't allow then to create or manage any functions from the Portal and point them to using VSCode instead. This will help those customers to setup their local projects with all necessary packages and devcontainer and they can easily add functions and then publish to azure from VSCode. We will have all this documented on our end on the Azure Portal as well as any documentations needed.

Reply:
Discussed in the meeting

Reply:
It sounded like C#Script was the priority for .NET in the portal, in which case you need to update your logic because we have a specific C#Script [ProjectLanguage](https://github.com/microsoft/vscode-azurefunctions/blob/main/src/constants.ts#L20). Also in that case, you _can_ download the app content. Other than that, I'm a little fuzzy on what we said for C# (aka _not_ script). What are we doing in that case? Sounded like y'all didn't care about that case, so I'd rather not deal with it in VS Code